### PR TITLE
refactor: prevent rapid polling calls from making multiple polling timers [DET-4183]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

The polling function creates multiple threads of polling if the polling function changes rapidly. One use case encountered that causes this is when we are change query parameters or filters for a list based endpoint. For example when we type in a search string in the search box for experiment list page.

## Test Plan

I don't know of a way to test for this without console logging inside the polling function. The easiest for a dev to test this is to put a `console.log('fetchExperiments', sorter, pagination, filters, search);` inside the `fetchExperimentList` function around line 129 and then type in the search box about 4 characters quickly and see the console logs popup in sets of 4 every 5 seconds. If this is fixed, this console log should be limited to 1 every 5 seconds.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Review by hamidzr — APPROVED** [review]
thanks for adding test instructions and fixing this problem! it works great.

**Comment by hamidzr** [comment]
would this need release note?

File: webui/react/src/hooks/usePolling.ts [hunk 1]
```diff
@@ -8,36 +8,42 @@ type PollingFn = (() => Promise<void>) | (() => void);
...
-    }, []);
-
-    useEffect(() => {
-      stopPolling();
-      pollingRoutine();
-      return stopPolling;
-    }, [ pollingRoutine, stopPolling ]);
+const usePolling = (pollingFn: PollingFn, { delay }: PollingOptions = {}): (() => void) => {
+  const timerId = useRef<NodeJS.Timeout>();
+  const countId = useRef(0);
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/hooks/usePolling.ts:15]: the `countId` was previously used to keep track of how many time the polling has occurred. More of an internal tracker.

[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/hooks/usePolling.ts:13]: I could've sworn the `clearTimeout` didn't work for me when I tried it first, but it's working great now... I'll update this.

[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/hooks/usePolling.ts]: So glad you asked this question because it's a lot cleaner now!

**Comment by hkang1** [comment]
> would this need release note?

yeah probably not, since not really customer facing.

File: webui/react/src/hooks/usePolling.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/react/src/hooks/usePolling.ts]: I'm glad it worked out!

**Review by hamidzr — APPROVED** [review]
LGTM :rocket:

**Pull request merged**